### PR TITLE
User test

### PR DESCRIPTION
**Pull request by andrlos**
~not working~ working user test :trollface:

### CODE REVIEW

File: src/test/java/cz/muni/pa165/teamwhite/formula1/UserTest.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+    @Autowired
+    private UserDao userDao;
+
+
+    @Test
+    public void testCreate(){
+        User kaja = new User("kaja", "CoTeToZajima", Role.MANAGER);
+        userDao.create(kaja);
+        Assert.assertTrue(true);
```
Comment (tomasedl) [inline src/test/java/cz/muni/pa165/teamwhite/formula1/UserTest.java:44]: :smile:

File: src/test/java/cz/muni/pa165/teamwhite/formula1/UserTest.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    }
+
+    @Test
+    @Rollback
+    public void testFetchByNameAndRemove(){
+        User tom = new User("lime", "4321", Role.ENGINEER);
+        userDao.create(tom);
+        tom = userDao.findByLogin("lime");
+        userDao.remove(tom);
+        Assert.assertThrows(NoResultException.class, () -> userDao.findByLogin("lime"));
```
Comment (tomasedl) [inline src/test/java/cz/muni/pa165/teamwhite/formula1/UserTest.java:64]: expectedExceptions?

File: src/test/java/cz/muni/pa165/teamwhite/formula1/UserTest.java (hunk 1 above)
Comment (Jakub-Fajkus) [reply src/test/java/cz/muni/pa165/teamwhite/formula1/UserTest.java:44]: Tests are passing boss! can i go home now?

File: src/test/java/cz/muni/pa165/teamwhite/formula1/UserTest.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.TestExecutionListeners;
+import org.springframework.test.context.testng.AbstractTestNGSpringContextTests;
+import org.springframework.test.context.transaction.TransactionalTestExecutionListener;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import javax.persistence.*;
```
Comment (Jakub-Fajkus) [inline src/test/java/cz/muni/pa165/teamwhite/formula1/UserTest.java:17]: Do not import everything, please... :cry:

File: src/test/java/cz/muni/pa165/teamwhite/formula1/UserTest.java [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+/**
+ * @author Jiří Andrlík
+ */
+@ContextConfiguration(classes = PersistenceSampleApplicationContext.class)
+@TestExecutionListeners(TransactionalTestExecutionListener.class)
+@Transactional
+public class UserTest extends AbstractTestNGSpringContextTests {
+
+    @PersistenceUnit
+    EntityManagerFactory emf;
```
Comment (Jakub-Fajkus) [inline src/test/java/cz/muni/pa165/teamwhite/formula1/UserTest.java:31]: unused

**Review by tomasedl — CHANGES_REQUESTED** [review]
same as Kuba, please add package for dao

File: src/test/java/cz/muni/pa165/teamwhite/formula1/UserTest.java [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+
+    @PersistenceContext
+    private EntityManager em;
```
Comment (Jakub-Fajkus) [inline src/test/java/cz/muni/pa165/teamwhite/formula1/UserTest.java:34]: unused as well

File: src/test/java/cz/muni/pa165/teamwhite/formula1/UserTest.java (hunk 3 above)
Comment (khecova) [inline src/test/java/cz/muni/pa165/teamwhite/formula1/UserTest.java:17]: jako ... asi by to tu byt nemelo :D "*"

File: src/test/java/cz/muni/pa165/teamwhite/formula1/UserTest.java (hunk 2 above)
Comment (Jakub-Fajkus) [reply src/test/java/cz/muni/pa165/teamwhite/formula1/UserTest.java:64]: I like this maybe even more than the annotation :thinking:... you are explicitly stating where you are expecting the exception...even if it is not as elegant as the annotation

File: src/test/java/cz/muni/pa165/teamwhite/formula1/UserTest.java (hunk 4 above)
Comment (khecova) [inline src/test/java/cz/muni/pa165/teamwhite/formula1/UserTest.java:31]: nevyuzita promenna

**Review by Jakub-Fajkus — CHANGES_REQUESTED** [review]

File: src/test/java/cz/muni/pa165/teamwhite/formula1/UserTest.java (hunk 2 above)
Comment (tomasedl) [reply src/test/java/cz/muni/pa165/teamwhite/formula1/UserTest.java:64]: ok, but at least we could use same "style" in one file

Comment (Jakub-Fajkus) [reply src/test/java/cz/muni/pa165/teamwhite/formula1/UserTest.java:64]: Definitelly...

Comment (andrlos) [reply src/test/java/cz/muni/pa165/teamwhite/formula1/UserTest.java]: solved :D

File: src/test/java/cz/muni/pa165/teamwhite/formula1/UserTest.java (hunk 1 above)
Comment (andrlos) [reply src/test/java/cz/muni/pa165/teamwhite/formula1/UserTest.java]: its a test on create... if it doesnt give an exception, I assume it passed :D

File: src/test/java/cz/muni/pa165/teamwhite/formula1/UserTest.java (hunk 3 above)
Comment (andrlos) [reply src/test/java/cz/muni/pa165/teamwhite/formula1/UserTest.java]: solved

File: src/test/java/cz/muni/pa165/teamwhite/formula1/UserTest.java (hunk 4 above)
Comment (andrlos) [reply src/test/java/cz/muni/pa165/teamwhite/formula1/UserTest.java]: removed

File: src/test/java/cz/muni/pa165/teamwhite/formula1/UserTest.java (hunk 5 above)
Comment (andrlos) [reply src/test/java/cz/muni/pa165/teamwhite/formula1/UserTest.java]: removed

File: src/test/java/cz/muni/pa165/teamwhite/formula1/dao/UserTest.java [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+
+    @Autowired
+    private UserDao userDao;
+
+
+    @Test
+    public void testCreate(){
+        User kaja = new User("kaja", "CoTeToZajima", Role.MANAGER);
+        userDao.create(kaja);
+        Assert.assertTrue(true);
```
Comment (Jakub-Fajkus) [inline src/test/java/cz/muni/pa165/teamwhite/formula1/dao/UserTest.java:34]: Could we just delete the assert? :smile: I cannot see any reason to keep it.

**Review by Jakub-Fajkus — APPROVED** [review]

File: src/test/java/cz/muni/pa165/teamwhite/formula1/UserTest.java (hunk 2 above)
Comment (Jakub-Fajkus) [reply src/test/java/cz/muni/pa165/teamwhite/formula1/UserTest.java]: :smile: did not expect that... but, why not :smile: Someday in the feature, you can rewrite all other tests to look this way :trollface:

**Review by tomasedl — APPROVED** [review]

**Review by khecova — APPROVED** [review]
nice 😄

**Pull request merged**